### PR TITLE
Task services - assets

### PR DESCRIPTION
**Pull request by laics1984**
- UX
- load existing assets in asset supply change

### CODE REVIEW

File: src/components/SupplyInput.vue [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
-      <input v-else-if="decimal==5" v-maska="'#*.#####'" :disabled="disabled == 1"  class="text-placeholder bg-white text-right" :value="parseFloat(modelValue)" @input="$emit('update:modelValue', parseFloat($event.target.value))" :placeholder="placeholder" @click="clickInputText()" @keyup="checkBalance($event)" @focus="$event.target.select()" @blur="blurInputText()">
-      <input v-else v-maska="'#*.######'" :disabled="disabled == 1"  class="text-placeholder bg-white text-right" :value="parseFloat(modelValue)" @input="$emit('update:modelValue', parseFloat($event.target.value))" :placeholder="placeholder" @click="clickInputText()" @keyup="checkBalance($event)" @focus="$event.target.select()" @blur="blurInputText()">
+      <input v-if="decimal==0" v-maska="'#*'" :disabled="disabled == true" class="text-placeholder bg-white text-right" :value="modelValue" @input="$emit('update:modelValue', $event.target.value)" :placeholder="placeholder" @click="clickInputText()" @keyup="checkBalance($event)" @focus="$event.target.select()" @blur="blurInputText()">
```
Comment (sleepyOwl14) [inline src/components/SupplyInput.vue:8]: `disabled == true`, is it not same if just put `disabled` as expression ?

File: src/util/assetsUtils.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  static getTransactionFee =(networkType: NetworkType, generationHash: string, owner:PublicAccount, supplyMutable: boolean, transferable:boolean, divisibility: number, duration: number) :number => {
+
+    const buildTransactions = new BuildTransactions(networkType, generationHash);
+    // owner: PublicAccount, supplyMutable: boolean, transferable: boolean, divisibility: number, duration?: UInt64
+    const registerMosaicTransaction = buildTransactions.mosaicDefinition(owner, supplyMutable, transferable, divisibility, UInt64.fromUint(duration));
+    return registerMosaicTransaction.maxFee.compact();
+  };
+
+  static getMosaicSupplyChangeTransaction = (networkType: NetworkType, generationHash: string, owner: PublicAccount, mosaicId: MosaicId, changeType: number, delta: UInt64) => {
```
Comment (sleepyOwl14) [inline src/util/assetsUtils.ts:56]: getMosaicSupplyChangeTransaction -> getMosaicSupplyChangeTransactionFee

File: src/util/assetsUtils.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    const buildTransactions = new BuildTransactions(networkType, generationHash);
+    let supplyChangeType: MosaicSupplyType;
+    supplyChangeType = (changeType==1)?MosaicSupplyType.Increase:MosaicSupplyType.Decrease;
+    const registerMosaicSupplyChangeTransaction = buildTransactions.buildMosaicSupplyChange(mosaicId, supplyChangeType, delta);
+    return registerMosaicSupplyChangeTransaction.maxFee.compact();
+  };
+
+  static getAssets = (account: WalletAccount, owner: string) => {
```
Comment (sleepyOwl14) [inline src/util/assetsUtils.ts:64]: getAssets -> getOwnedAssets

File: src/util/assetsUtils.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    const assetSelection: Array<assetSelectionInterface> = [];
+    // walletState.currentLoggedInWallet.selectDefaultAccount().publicKey
+    const filterAsset = account.assets.filter((asset) => asset.owner === owner);
+    if(filterAsset.length > 0){
+      filterAsset.forEach((asset) => {
+        assetSelection.push({
+          label: asset.idHex + ' > ' + Helper.amountFormatterSimple(asset.amount, asset.divisibility) + ' XPX',
```
Comment (sleepyOwl14) [inline src/util/assetsUtils.ts:70]: the amount already convert to exact amount, no need to convert again.
please remove the `XPX` string, asset does not have name by default

File: src/modules/services/submodule/assets/views/ViewServicesAssetsModifySupplyChange.vue [hunk 5]
```diff
@@ -156,23 +170,56 @@ export default {
...
+
     const assetOptions = () => {
-      let asset = [];
-      return asset;
+      let assetSelection = [];
+      const filterAsset = walletState.currentLoggedInWallet.selectDefaultAccount().assets.filter((asset) => asset.owner === walletState.currentLoggedInWallet.selectDefaultAccount().publicKey);
+      if(filterAsset.length > 0){
+        filterAsset.forEach((asset) => {
+          assetSelection.push({
+            label: asset.idHex + ' > ' + Helper.amountFormatterSimple(asset.amount, asset.divisibility) + ' XPX',
```
Comment (sleepyOwl14) [inline src/modules/services/submodule/assets/views/ViewServicesAssetsModifySupplyChange.vue:217]: not need to convert asset amount, remove `XPX`

**Review by sleepyOwl14 — CHANGES_REQUESTED** [review]
please fix, thanks

File: src/util/assetsUtils.ts (hunk 2 above)
Comment (sleepyOwl14) [reply src/util/assetsUtils.ts:56]: no, not that 1. Just change the static method name

Comment (laics1984) [reply src/util/assetsUtils.ts:56]: Okay, got it. Have updated method for mosaic definiteion in assetsCreate vue file too

File: src/util/assetsUtils.ts (hunk 4 above)
Comment (laics1984) [reply src/util/assetsUtils.ts:70]: Noted on the XPX. However for the exact amount, I am displaying the amount in relative to its divisibility. Like how it appears on the angular wallet. Pls advice. Thanks. 

image

File: src/modules/services/submodule/assets/views/ViewServicesAssetsModifySupplyChange.vue (hunk 5 above)
Comment (laics1984) [reply src/modules/services/submodule/assets/views/ViewServicesAssetsModifySupplyChange.vue:217]: Okay noted with thanks

**Comment by alvin-reyes** [comment]
Task linked: [CU-9qz7x5 Sub module assets](https://app.clickup.com/t/9qz7x5)

File: src/modules/services/submodule/assets/views/ViewServicesAssetsCreate.vue [hunk 6]
```diff
@@ -284,6 +326,50 @@ export default {
...
+        disabledMutableCheck.value = false;
+        disabledTransferableCheck.value = false;
+        disabledPassword.value = false;
+        disabledSupply.value = false;
+        disabledDivisibility.value = false;
+      }
+    });
+
     const createMosaic = () => {
       let createStatus = mosaicTransaction(divisibility.value, supply.value, duration.value, durationOption.value, isMutable.value, isTransferable.value, walletPassword.value, selectedAccName.value, appStore, siriusStore);
```
Comment (sleepyOwl14) [inline src/modules/services/submodule/assets/views/ViewServicesAssetsCreate.vue:374]: this still using ? there are using appStore and siriusStore

File: src/modules/services/submodule/assets/views/ViewServicesAssetsModifySupplyChange.vue [hunk 7]
```diff
@@ -111,70 +146,101 @@ export default {
...
+    const assetSupply = ref(0);
+    const assetAmount = ref(0);
+    const assetDuration = ref('0 Day');
+    const assetDivisibility = ref(0);
+    const assetTransferable = ref(false);
+    const assetMutable = ref(false);
+
+    const changeAsset = (assetId) => {
+      const selectedAsset = walletState.currentLoggedInWallet.selectDefaultAccount().assets.find((asset) => asset.idHex === assetId);
+      assetSupply.value =Helper.amountFormatterSimple(selectedAsset.amount, selectedAsset.divisibility);
```
Comment (sleepyOwl14) [inline src/modules/services/submodule/assets/views/ViewServicesAssetsModifySupplyChange.vue:226]: variable is assetSupply, but you use the asset amount value

**Review by sleepyOwl14 — CHANGES_REQUESTED** [review]
please find issue from conversation

File: src/modules/services/submodule/assets/views/ViewServicesAssetsCreate.vue (hunk 6 above)
Comment (laics1984) [reply src/modules/services/submodule/assets/views/ViewServicesAssetsCreate.vue]: Ops. This is no longer needed. To be replaced with create asset method once it is ready

File: src/modules/services/submodule/assets/views/ViewServicesAssetsModifySupplyChange.vue (hunk 7 above)
Comment (laics1984) [reply src/modules/services/submodule/assets/views/ViewServicesAssetsModifySupplyChange.vue]: Thanks, have corrected the property

File: src/modules/services/submodule/assets/views/ViewServicesAssetsCreate.vue (hunk 6 above)
Comment (sleepyOwl14) [reply src/modules/services/submodule/assets/views/ViewServicesAssetsCreate.vue]: ok, maybe put some comment on it

**Review by sleepyOwl14 — APPROVED** [review]

**Comment by laics1984** [comment]
Summary:
- UX for sub module assets - create new asset and change supply vue
- Fetch available assets for each account in change supply account
- validate cosignership of multisig account
- validate if account has assets in change supply vue
- Bug fixes on reloading assets when account is changed

File: src/modules/services/submodule/assets/views/ViewServicesAssetsCreate.vue (hunk 6 above)
Comment (laics1984) [reply src/modules/services/submodule/assets/views/ViewServicesAssetsCreate.vue]: Okay noted

**Pull request merged**